File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        priority_queue<pair<int, int>, vector<pair<int, int>>> frequency;
        for (int i=0; i<mat.size(); i++) {
            frequency.push(accumulate(mat[i].begin(), mat[i].end(),0));
        }
        vector<int> answer;
        for (int i=0; i<k; i++) {
            auto p = frequency.top();
            frequency.pop();
            answer.push_back(p[1]);
        } return answer;
    }
};

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.py
from collections import defaultdict
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        frequency = []
        for i in range(len(mat)):
            frequency.append((sum(mat[i]), i))
        heapq.heapify(frequency)
        return [heapq.heappop(frequency)[1] for _ in range(k)]
        # frequency.sort(key=lambda x:(x[0],x[1]))
        # answer = []
        # for f in frequency:
        #     answer.append(f[1])
        # return answer[:k]